fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -4014,6 +4014,13 @@ public void alterTableProperties(Database db, OlapTable table, Map<String, Strin
         if (logProperties.containsKey(PropertyAnalyzer.PROPERTIES_PARTITION_LIVE_NUMBER)) {
```

Comment:
This `logProperties.containsKey(PropertyAnalyzer.PROPERTIES_PARTITION_LIVE_NUMBER)` has been checked twice, why not put them together?

Reply:
When using table properties, analyzeXXX will analyze and remove the XXX properties. This is why you need to copy it.

Therefore, before modifying the memory, we will first check whether the attributes are legal and identifiable. If they are put together, it means that the memory will be modified after the check is completed. It is possible that after other attributes are judged, it will be found that the entire statement is actually is illegal. For example, the user writes a property that does not exist.

At this time, you have to roll back the previous memory modifications. Therefore, the code looks like it does a check above, and then checks the corresponding attributes for memory modification below. It takes 2 times.

Reply:
Understood, but the logic is quite confusing, how about moving the code between L3987-L4013 to a new function like `validateToBeModifiedProps()` and this func returns a analyzed result map `Map<String, Object> results`. Then in `alterTableProperties()`, iterate that `results` map and modify the real table property based on the map key coorespondingly.